lib/puppet/module.rb

```diff
@@ -214,7 +214,8 @@ def license_file
 
   def read_metadata
     md_file = metadata_file
-    md_file.nil? ? {} : Puppet::Util::Json.load(File.read(md_file, :encoding => 'utf-8'))
+    content = File.read(md_file, :encoding => 'utf-8')
+    content.empty? ? {} : Puppet::Util::Json.load(content)
```

Comment:
I think we still need the check for `md_file.nil?` and return `{}` in that case.